### ISSUE TITLE
ConnectedToUser not called if call initiated from iOS to Android

### ISSUE THREAD

**Issue by devsherif** [QuickBlox/quickblox-ios-sdk#600]
_Help avoid duplicate issue reports, check [existing issues](https://github.com/QuickBlox/quickblox-ios-sdk/issues)_

**Environment details**
_iOS version, Quickblox iOS SDK version, QuickbloxWebRTC SDK version (optional)_
iOS Version :9.3, QuickBlox : 2.7.6, QuickBloxWebRTC : 2.2

**Did this work before?**
NO

**Expected behavior**
After Call Accepted by user, ConnectedToUser should be called

**Actual behavior**
After Call Accepted By User, ConnectedToUser should be called , but nothing happens until the call times out

**Logs**
_Attach full logs that will describe your problem (please, use github [gist](https://gist.github.com/))_
https://gist.github.com/devsherif/16c8d70185de12cd25639be5de54532d

**Steps to reproduce the behavior**
Just Initiate the call from an iOS device to an Android Device
Or from iOS device to iOS device

**Any others comments?**
Just wanted to mention that with the same code, if call initiated from Android to iOS device, it works normally.

**Comment by Pro100Andrey** [comment]
Hi @devsherif, please provide logs from both devices.

**Comment by Pro100Andrey** [comment]
@devsherif and please check - Android [Accept call] https://quickblox.com/developers/Sample-webrtc-android#Accept_call

**Comment by devsherif** [comment]
@Pro100Andrey , regarding "Accept Call". As I mentioned Android to Android calls work perfectly, also Android to iOS. So I use accept call and it works, video call is completely established.
So the issue take place only if the call is initiated from iOS either to iOS or Android.
From Android I see the "Accept Call" action is sent. 

Extracted logs for full call attempt session from iOS to Android could be found here:
https://gist.github.com/devsherif/a460dcd6dd21939b409a7cfb14f5c95c
This session also call is initiated by the iOS device, Android started ringing, Accept call action is sent from Android to iOS, iOS received call acceptance as method "acceptedByUser" is called, then it keeps in "Connecting" state and method "connectedToUser" is never called.

**Comment by waterweaver** [comment]
hi me also have the same problem. any updates ?

**Comment by Pro100Andrey** [comment]
Hi @devsherif Please provide logs from both devices (iOS, Android) for the  some SessionID

**Comment by Pro100Andrey** [comment]
@devsherif  Any updates?

**Comment by Pro100Andrey** [comment]
Hi @devsherif, 

ios - F4D80303-D889-4A43-8417-A186C9FC4EE6 session ID
android -  F4D80303-D889-4A43-8417-A186C9FC4EE6 session ID

**Comment by Pro100Andrey** [comment]
Hi @devsherif, from you logs:
ios Session ID - B9C9EC94-B8DF-4E98-8A90-E1926DD4092B
android Session ID - F4D80303-D889-4A43-8417-A186C9FC4EE6

**Comment by devsherif** [comment]
Hi @Pro100Andrey :
I did the same test again, and found in my logs that Session ID is the same on Android and iOS.
One note: Do you think if session ID is different, the iphone will start the call and android receive it then iOS call status changes to be "Connecting" !!! I believe the session ID is the same on both android and iOS.
Anyways, I attached the new logs. In which the same behaviour took place, iOS initiated the call, Android received then accepted it, iOS "startedConnectingToUser" delegate method is called, then nothing more happened, call dropped.

Kindly find attached the new logs
[ANDROID.TXT](https://github.com/QuickBlox/quickblox-ios-sdk/files/521153/ANDROID.TXT)
[IOS.TXT](https://github.com/QuickBlox/quickblox-ios-sdk/files/521154/IOS.TXT)

**Comment by Pro100Andrey** [comment]
Hi @devsherif, we are not able to reproduce this problem in our samples. Can you confirm this issue in our samples:

+ [iOS sample](https://github.com/QuickBlox/quickblox-ios-sdk/tree/master/sample-videochat-webrtc)

+ [Android sample](https://github.com/QuickBlox/quickblox-android-sdk/tree/master/sample-videochat-webrtc)

**Comment by VitaliyGuru** [comment]
@devsherif  Any updates?

**Comment by devsherif** [comment]
Hi @Pro100Andrey , @VitaliyGuru ,
Sorry for my late reply.
Using the samples, the issue is not reproducible.
But even after following the samples, I still couldn't find the issue in my code. I didn't have much time to investigate my code yet, but will do soon and let you know what I will find.

**Comment by Raikerian** [comment]
Hi @devsherif 

Were you able to fix this?

**Comment by Raikerian** [comment]
Closing due to lack of user response for now. If the issue is still there for you, create a new one. Attaching sample project with issue been reproducible is the ideal description.

**Comment by devsherif** [comment]
Hi @Raikerian 

Finally I was able to find the root cause of the issue and fix it. Let me explain it as It may need an update in QuickBlox code.

Briefly, the issue happened because I create a new Session in ViewController then pass it to another ViewController.

So, noting that both ViewControllers are QBRTCClientDelegate, it seems very miss-leading to get most of overridden method being called normally except for "StartedConnectingToUser".
Once I moved "createNewSessionWithOpponents" call to the second ViewController, everything worked perfectly.

Thanks for your support. And I hope that I have explained the issue properly and clearly.
